beekeeper-path-scheduler-apiary/src/main/java/com/expediagroup/beekeeper/scheduler/apiary/handler/MessageEventHandler.java

```diff
@@ -78,37 +77,22 @@ private boolean shouldFilterMessage(ListenerEvent listenerEvent) {
    * @param listenerEvent Listener event from the current message
    * @return list of housekeeping paths. This can be an empty list if there are no valid paths.
    */
-  private List<HousekeepingPath> generateHouseKeepingPaths(ListenerEvent listenerEvent) {
+  private List<Housekeeping> generateHousekeepingEntities(ListenerEvent listenerEvent) {
```

Comment:
this method + the two abstract methods it calls are complex and flow is hard to follow, the generics don't help. EventModel implementations aren't really used anywhere except in the concrete subclasses and are unnecessary maybe. Perhaps we can instantiate this class with HouseKeepingEntityGenerator class that can turn do ```List< Housekeeping> generate(listenerEvent)```. Encapsulating the two abstract methods which can be removed.
Then can have two implementations (similar to the two subclasses we have now) that way we move the inheritance to composition which will be easier to test.